independent-publisher-connectors/HubSpot CRM/apiProperties.json

```diff
@@ -0,0 +1,75 @@
+{
+  "properties": {
+    "connectionParameterSets": {
+      "uiDefinition": {
+        "displayName": "Authentication Type",
+        "description": "Type of authentication to be used."
+      },
+      "values": [
+        {
+          "name": "hubspot-oauth",
+          "uiDefinition": {
+            "displayName": "Use HubSpot OAuth 2.0",
+            "description": "Log in using HubSpot app client ID and secret."
+          },
+          "parameters": {
+            "token": {
+              "type": "oauthSetting",
+              "oAuthSettings": {
+                "identityProvider": "oauth2",
+                "clientId": "[Your_Value]",
+                "scopes": [
+                  "contacts oauth"
+                ],
+                "redirectMode": "Global",
+                "redirectUrl": "https://global.consent.azure-apim.net/redirect",
+                "properties": {
+                  "IsFirstParty": "False"
+                },
+                "customParameters": {
+                  "authorizationUrl": {
+                    "value": "https://app.hubspot.com/oauth/authorize"
+                  },
+                  "tokenUrl": {
+                    "value": "https://api.hubapi.com/oauth/v1/token"
+                  },
+                  "refreshUrl": {
+                    "value": "https://api.hubapi.com/oauth/v1/token"
+                  }
+                }
+              }
+            }
+          }
+        },
+        {
+          "name": "api-auth",
+          "uiDefinition": {
+            "displayName": "Use HubSpot API Key",
+            "description": "Log in using a HubSpot API Key."
+          },
+          "parameters": {
+            "api_key": {

```

Comment:
According to the HubSpot documentation:
https://legacydocs.hubspot.com/docs/methods/auth/oauth-overview
```
API keys are great for rapid prototyping, but for security and commercial use, all integrations designed for use by multiple HubSpot customers should use OAuth—this is a requirement to be listed in our App Marketplace.
```
Also the api key is named as `hapikey` and provided as part of the query parameter. Please either remove api key auth or update this connection method. Please test your changes before updating. Please make similar change for all HubSpot connectors.

Reply:
Fixed all properties to hapikey. HubSpot documentation is only providing security guidance - some companies may be comfortable with using an API key for their security. I don't expect any commercial users of these custom connectors to list them on the marketplace. Additional direction from Natalie requested as multi-auth was required for these connectors.